### ISSUE TITLE
Failure when training Imagenet Example from tutorial

### ISSUE THREAD

**Issue by andersonRocha091** [BVLC/caffe#2706]
Hi, I was going through the imagenet tutorial and I just finished running the script make_imagenet_mean.sh. However, when I was trying to train the network according to the author's tutorial through the command ./build/tools/caffe train --solver=models/bvlc_reference_caffenet/solver.prototxt but I got the following message when caffe was generating the test network: 

I0706 11:00:31.481516  8842 layer_factory.hpp:74] Creating layer data
I0706 11:00:31.481535  8842 net.cpp:90] Creating Layer data
I0706 11:00:31.481547  8842 net.cpp:368] data -> data
I0706 11:00:31.481564  8842 net.cpp:368] data -> label
I0706 11:00:31.481578  8842 net.cpp:120] Setting up data
I0706 11:00:31.481590  8842 data_transformer.cpp:22] Loading mean file from: data/ilsvrc12/imagenet_mean.binaryproto
F0706 11:00:31.485407  8842 db_lmdb.hpp:13] Check failed: mdb_status == 0 (2 vs. 0) No such file or directory
*** Check failure stack trace: ***
    @     0x7f906fb6bdaa  (unknown)
    @     0x7f906fb6bce4  (unknown)
    @     0x7f906fb6b6e6  (unknown)
    @     0x7f906fb6e687  (unknown)
    @     0x7f906ff9453e  caffe::db::LMDB::Open()
    @     0x7f906feff412  caffe::DataLayer<>::DataLayerSetUp()
    @     0x7f906ff33d19  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7f906ffafe33  caffe::Net<>::Init()
    @     0x7f906ffb1ba2  caffe::Net<>::Net()
    @     0x7f906ff6e034  caffe::Solver<>::InitTestNets()
    @     0x7f906ff6e72b  caffe::Solver<>::Init()
    @     0x7f906ff6e8f6  caffe::Solver<>::Solver()
    @           0x40c970  caffe::GetSolver<>()
    @           0x406621  train()
    @           0x404ab1  main
    @     0x7f906f07dec5  (unknown)
    @           0x40505d  (unknown)
    @              (nil)  (unknown)

I have tried to find the file db_lmdb.hpp but it is not on the caffe/src folder. Does anyone had the same problem?

here is the whole output that I got:

I0706 11:00:27.380079  8842 caffe.cpp:117] Use CPU.
I0706 11:00:27.380385  8842 caffe.cpp:121] Starting Optimization
I0706 11:00:27.380455  8842 solver.cpp:32] Initializing solver from parameters:
test_iter: 1000
test_interval: 1000
base_lr: 0.01
display: 20
max_iter: 450000
lr_policy: "step"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
stepsize: 100000
snapshot: 10000
snapshot_prefix: "models/bvlc_reference_caffenet/caffenet_train"
solver_mode: CPU
net: "models/bvlc_reference_caffenet/train_val.prototxt"
I0706 11:00:27.380480  8842 solver.cpp:70] Creating training net from net file: models/bvlc_reference_caffenet/train_val.prototxt
I0706 11:00:27.381482  8842 net.cpp:287] The NetState phase (0) differed from the phase (1) specified by a rule in layer data
I0706 11:00:27.381518  8842 net.cpp:287] The NetState phase (0) differed from the phase (1) specified by a rule in layer accuracy
I0706 11:00:27.381767  8842 net.cpp:42] Initializing net from parameters:
name: "CaffeNet"
state {
  phase: TRAIN
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 227
    mean_file: "data/ilsvrc12/imagenet_mean.binaryproto"
  }
 data_param {
    source: "examples/imagenet/ilsvrc12_train_lmdb"
    batch_size: 256
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
 }
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "pool1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "pool2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
 beta: 0.75
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "norm2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
  top: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
 lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
  bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
 dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
  bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1000
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
I0706 11:00:27.381924  8842 layer_factory.hpp:74] Creating layer data
I0706 11:00:27.381959  8842 net.cpp:90] Creating Layer data
I0706 11:00:27.381973  8842 net.cpp:368] data -> data
I0706 11:00:27.382009  8842 net.cpp:368] data -> label
I0706 11:00:27.382028  8842 net.cpp:120] Setting up data
I0706 11:00:27.382040  8842 data_transformer.cpp:22] Loading mean file from: data/ilsvrc12/imagenet_mean.binaryproto
I0706 11:00:27.386047  8842 db_lmdb.cpp:22] Opened lmdb examples/imagenet/ilsvrc12_train_lmdb
I0706 11:00:27.386396  8842 data_layer.cpp:52] output data size: 256,3,227,227
I0706 11:00:27.607839  8842 net.cpp:127] Top shape: 256 3 227 227 (39574272)
I0706 11:00:27.607930  8842 net.cpp:127] Top shape: 256 (256)
I0706 11:00:27.607951  8842 layer_factory.hpp:74] Creating layer conv1
I0706 11:00:27.607988  8842 net.cpp:90] Creating Layer conv1
I0706 11:00:27.608003  8842 net.cpp:410] conv1 <- data
I0706 11:00:27.608028  8842 net.cpp:368] conv1 -> conv1
I0706 11:00:27.608057  8842 net.cpp:120] Setting up conv1
I0706 11:00:27.610326  8842 net.cpp:127] Top shape: 256 96 55 55 (74342400)
I0706 11:00:27.610348  8842 layer_factory.hpp:74] Creating layer relu1
I0706 11:00:27.610365  8842 net.cpp:90] Creating Layer relu1
I0706 11:00:27.610375  8842 net.cpp:410] relu1 <- conv1
I0706 11:00:27.610388  8842 net.cpp:357] relu1 -> conv1 (in-place)
I0706 11:00:27.610401  8842 net.cpp:120] Setting up relu1
I0706 11:00:27.610422  8842 net.cpp:127] Top shape: 256 96 55 55 (74342400)
I0706 11:00:27.610436  8842 layer_factory.hpp:74] Creating layer pool1
I0706 11:00:27.610451  8842 net.cpp:90] Creating Layer pool1
I0706 11:00:27.610460  8842 net.cpp:410] pool1 <- conv1
I0706 11:00:27.610473  8842 net.cpp:368] pool1 -> pool1
I0706 11:00:27.610488  8842 net.cpp:120] Setting up pool1
I0706 11:00:27.610513  8842 net.cpp:127] Top shape: 256 96 27 27 (17915904)
I0706 11:00:27.610524  8842 layer_factory.hpp:74] Creating layer norm1
I0706 11:00:27.610540  8842 net.cpp:90] Creating Layer norm1
I0706 11:00:27.610550  8842 net.cpp:410] norm1 <- pool1
I0706 11:00:27.610563  8842 net.cpp:368] norm1 -> norm1
I0706 11:00:27.610580  8842 net.cpp:120] Setting up norm1
I0706 11:00:27.610596  8842 net.cpp:127] Top shape: 256 96 27 27 (17915904)
I0706 11:00:27.610606  8842 layer_factory.hpp:74] Creating layer conv2
I0706 11:00:27.610646  8842 net.cpp:90] Creating Layer conv2
I0706 11:00:27.610677  8842 net.cpp:410] conv2 <- norm1
I0706 11:00:27.610692  8842 net.cpp:368] conv2 -> conv2
I0706 11:00:27.610704  8842 net.cpp:120] Setting up conv2
I0706 11:00:27.630013  8842 net.cpp:127] Top shape: 256 256 27 27 (47775744)
I0706 11:00:27.630046  8842 layer_factory.hpp:74] Creating layer relu2
I0706 11:00:27.630061  8842 net.cpp:90] Creating Layer relu2
I0706 11:00:27.630070  8842 net.cpp:410] relu2 <- conv2
I0706 11:00:27.630084  8842 net.cpp:357] relu2 -> conv2 (in-place)
I0706 11:00:27.630096  8842 net.cpp:120] Setting up relu2
I0706 11:00:27.630108  8842 net.cpp:127] Top shape: 256 256 27 27 (47775744)
I0706 11:00:27.630117  8842 layer_factory.hpp:74] Creating layer pool2
I0706 11:00:27.630131  8842 net.cpp:90] Creating Layer pool2
I0706 11:00:27.630139  8842 net.cpp:410] pool2 <- conv2
I0706 11:00:27.630151  8842 net.cpp:368] pool2 -> pool2
I0706 11:00:27.630163  8842 net.cpp:120] Setting up pool2
I0706 11:00:27.630178  8842 net.cpp:127] Top shape: 256 256 13 13 (11075584)
I0706 11:00:27.630188  8842 layer_factory.hpp:74] Creating layer norm2
I0706 11:00:27.630204  8842 net.cpp:90] Creating Layer norm2
I0706 11:00:27.630214  8842 net.cpp:410] norm2 <- pool2
I0706 11:00:27.630226  8842 net.cpp:368] norm2 -> norm2
I0706 11:00:27.630239  8842 net.cpp:120] Setting up norm2
I0706 11:00:27.630252  8842 net.cpp:127] Top shape: 256 256 13 13 (11075584)
I0706 11:00:27.630261  8842 layer_factory.hpp:74] Creating layer conv3
I0706 11:00:27.630277  8842 net.cpp:90] Creating Layer conv3
I0706 11:00:27.630286  8842 net.cpp:410] conv3 <- norm2
I0706 11:00:27.630300  8842 net.cpp:368] conv3 -> conv3
I0706 11:00:27.630314  8842 net.cpp:120] Setting up conv3
I0706 11:00:27.685798  8842 net.cpp:127] Top shape: 256 384 13 13 (16613376)
I0706 11:00:27.685876  8842 layer_factory.hpp:74] Creating layer relu3
I0706 11:00:27.685895  8842 net.cpp:90] Creating Layer relu3
I0706 11:00:27.685906  8842 net.cpp:410] relu3 <- conv3
I0706 11:00:27.685920  8842 net.cpp:357] relu3 -> conv3 (in-place)
I0706 11:00:27.685935  8842 net.cpp:120] Setting up relu3
I0706 11:00:27.685948  8842 net.cpp:127] Top shape: 256 384 13 13 (16613376)
I0706 11:00:27.685957  8842 layer_factory.hpp:74] Creating layer conv4
I0706 11:00:27.685973  8842 net.cpp:90] Creating Layer conv4
I0706 11:00:27.685982  8842 net.cpp:410] conv4 <- conv3
I0706 11:00:27.685995  8842 net.cpp:368] conv4 -> conv4
I0706 11:00:27.686009  8842 net.cpp:120] Setting up conv4
I0706 11:00:27.727751  8842 net.cpp:127] Top shape: 256 384 13 13 (16613376)
I0706 11:00:27.727814  8842 layer_factory.hpp:74] Creating layer relu4
I0706 11:00:27.727833  8842 net.cpp:90] Creating Layer relu4
I0706 11:00:27.727843  8842 net.cpp:410] relu4 <- conv4
I0706 11:00:27.727857  8842 net.cpp:357] relu4 -> conv4 (in-place)
I0706 11:00:27.727871  8842 net.cpp:120] Setting up relu4
I0706 11:00:27.727885  8842 net.cpp:127] Top shape: 256 384 13 13 (16613376)
I0706 11:00:27.727893  8842 layer_factory.hpp:74] Creating layer conv5
I0706 11:00:27.727910  8842 net.cpp:90] Creating Layer conv5
I0706 11:00:27.727919  8842 net.cpp:410] conv5 <- conv4
I0706 11:00:27.727932  8842 net.cpp:368] conv5 -> conv5
I0706 11:00:27.727946  8842 net.cpp:120] Setting up conv5
I0706 11:00:27.755506  8842 net.cpp:127] Top shape: 256 256 13 13 (11075584)
I0706 11:00:27.755568  8842 layer_factory.hpp:74] Creating layer relu5
I0706 11:00:27.755587  8842 net.cpp:90] Creating Layer relu5
I0706 11:00:27.755597  8842 net.cpp:410] relu5 <- conv5
I0706 11:00:27.755611  8842 net.cpp:357] relu5 -> conv5 (in-place)
I0706 11:00:27.755625  8842 net.cpp:120] Setting up relu5
I0706 11:00:27.755637  8842 net.cpp:127] Top shape: 256 256 13 13 (11075584)
I0706 11:00:27.755647  8842 layer_factory.hpp:74] Creating layer pool5
I0706 11:00:27.755661  8842 net.cpp:90] Creating Layer pool5
I0706 11:00:27.755671  8842 net.cpp:410] pool5 <- conv5
I0706 11:00:27.755683  8842 net.cpp:368] pool5 -> pool5
I0706 11:00:27.755697  8842 net.cpp:120] Setting up pool5
I0706 11:00:27.755714  8842 net.cpp:127] Top shape: 256 256 6 6 (2359296)
I0706 11:00:27.755769  8842 layer_factory.hpp:74] Creating layer fc6
I0706 11:00:27.755807  8842 net.cpp:90] Creating Layer fc6
I0706 11:00:27.755817  8842 net.cpp:410] fc6 <- pool5
I0706 11:00:27.755831  8842 net.cpp:368] fc6 -> fc6
I0706 11:00:27.755852  8842 net.cpp:120] Setting up fc6
I0706 11:00:30.161845  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:30.161902  8842 layer_factory.hpp:74] Creating layer relu6
I0706 11:00:30.161921  8842 net.cpp:90] Creating Layer relu6
I0706 11:00:30.161932  8842 net.cpp:410] relu6 <- fc6
I0706 11:00:30.161947  8842 net.cpp:357] relu6 -> fc6 (in-place)
I0706 11:00:30.161962  8842 net.cpp:120] Setting up relu6
I0706 11:00:30.161973  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:30.161983  8842 layer_factory.hpp:74] Creating layer drop6
I0706 11:00:30.161998  8842 net.cpp:90] Creating Layer drop6
I0706 11:00:30.162006  8842 net.cpp:410] drop6 <- fc6
I0706 11:00:30.162019  8842 net.cpp:357] drop6 -> fc6 (in-place)
I0706 11:00:30.162036  8842 net.cpp:120] Setting up drop6
I0706 11:00:30.162053  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:30.162062  8842 layer_factory.hpp:74] Creating layer fc7
I0706 11:00:30.162077  8842 net.cpp:90] Creating Layer fc7
I0706 11:00:30.162086  8842 net.cpp:410] fc7 <- fc6
I0706 11:00:30.162099  8842 net.cpp:368] fc7 -> fc7
I0706 11:00:30.162114  8842 net.cpp:120] Setting up fc7
I0706 11:00:31.220427  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:31.220481  8842 layer_factory.hpp:74] Creating layer relu7
I0706 11:00:31.220500  8842 net.cpp:90] Creating Layer relu7
I0706 11:00:31.220511  8842 net.cpp:410] relu7 <- fc7
I0706 11:00:31.220525  8842 net.cpp:357] relu7 -> fc7 (in-place)
I0706 11:00:31.220540  8842 net.cpp:120] Setting up relu7
I0706 11:00:31.220552  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:31.220561  8842 layer_factory.hpp:74] Creating layer drop7
I0706 11:00:31.220576  8842 net.cpp:90] Creating Layer drop7
I0706 11:00:31.220584  8842 net.cpp:410] drop7 <- fc7
I0706 11:00:31.220597  8842 net.cpp:357] drop7 -> fc7 (in-place)
I0706 11:00:31.220608  8842 net.cpp:120] Setting up drop7
I0706 11:00:31.220623  8842 net.cpp:127] Top shape: 256 4096 (1048576)
I0706 11:00:31.220631  8842 layer_factory.hpp:74] Creating layer fc8
I0706 11:00:31.220646  8842 net.cpp:90] Creating Layer fc8
I0706 11:00:31.220655  8842 net.cpp:410] fc8 <- fc7
I0706 11:00:31.220669  8842 net.cpp:368] fc8 -> fc8
I0706 11:00:31.220684  8842 net.cpp:120] Setting up fc8
I0706 11:00:31.478379  8842 net.cpp:127] Top shape: 256 1000 (256000)
I0706 11:00:31.478432  8842 layer_factory.hpp:74] Creating layer loss
I0706 11:00:31.478451  8842 net.cpp:90] Creating Layer loss
I0706 11:00:31.478461  8842 net.cpp:410] loss <- fc8
I0706 11:00:31.478472  8842 net.cpp:410] loss <- label
I0706 11:00:31.478490  8842 net.cpp:368] loss -> loss
I0706 11:00:31.478505  8842 net.cpp:120] Setting up loss
I0706 11:00:31.478520  8842 layer_factory.hpp:74] Creating layer loss
I0706 11:00:31.479446  8842 net.cpp:127] Top shape: (1)
I0706 11:00:31.479468  8842 net.cpp:129]     with loss weight 1
I0706 11:00:31.479502  8842 net.cpp:192] loss needs backward computation.
I0706 11:00:31.479513  8842 net.cpp:192] fc8 needs backward computation.
I0706 11:00:31.479522  8842 net.cpp:192] drop7 needs backward computation.
I0706 11:00:31.479532  8842 net.cpp:192] relu7 needs backward computation.
I0706 11:00:31.479540  8842 net.cpp:192] fc7 needs backward computation.
I0706 11:00:31.479550  8842 net.cpp:192] drop6 needs backward computation.
I0706 11:00:31.479559  8842 net.cpp:192] relu6 needs backward computation.
I0706 11:00:31.479568  8842 net.cpp:192] fc6 needs backward computation.
I0706 11:00:31.479578  8842 net.cpp:192] pool5 needs backward computation.
I0706 11:00:31.479588  8842 net.cpp:192] relu5 needs backward computation.
I0706 11:00:31.479598  8842 net.cpp:192] conv5 needs backward computation.
I0706 11:00:31.479606  8842 net.cpp:192] relu4 needs backward computation.
I0706 11:00:31.479615  8842 net.cpp:192] conv4 needs backward computation.
I0706 11:00:31.479662  8842 net.cpp:192] relu3 needs backward computation.
I0706 11:00:31.479672  8842 net.cpp:192] conv3 needs backward computation.
I0706 11:00:31.479681  8842 net.cpp:192] norm2 needs backward computation.
I0706 11:00:31.479691  8842 net.cpp:192] pool2 needs backward computation.
I0706 11:00:31.479701  8842 net.cpp:192] relu2 needs backward computation.
I0706 11:00:31.479709  8842 net.cpp:192] conv2 needs backward computation.
I0706 11:00:31.479718  8842 net.cpp:192] norm1 needs backward computation.
I0706 11:00:31.479727  8842 net.cpp:192] pool1 needs backward computation.
I0706 11:00:31.479737  8842 net.cpp:192] relu1 needs backward computation.
I0706 11:00:31.479746  8842 net.cpp:192] conv1 needs backward computation.
I0706 11:00:31.479755  8842 net.cpp:194] data does not need backward computation.
I0706 11:00:31.479764  8842 net.cpp:235] This network produces output loss
I0706 11:00:31.479792  8842 net.cpp:482] Collecting Learning Rate and Weight Decay.
I0706 11:00:31.479809  8842 net.cpp:247] Network initialization done.
I0706 11:00:31.479817  8842 net.cpp:248] Memory required for data: 1757220868
I0706 11:00:31.480965  8842 solver.cpp:154] Creating test net (#0) specified by net file: models/bvlc_reference_caffenet/train_val.prototxt
I0706 11:00:31.481032  8842 net.cpp:287] The NetState phase (1) differed from the phase (0) specified by a rule in layer data
I0706 11:00:31.481326  8842 net.cpp:42] Initializing net from parameters:
name: "CaffeNet"
state {
  phase: TEST
}
layer {
  name: "data"
  type: "Data"
  top: "data"
  top: "label"
include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 227
    mean_file: "data/ilsvrc12/imagenet_mean.binaryproto"
  }
  data_param {
    source: "examples/imagenet/ilsvrc12_val_lmdb"
    batch_size: 50
    backend: LMDB
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "data"
  top: "conv1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 96
    kernel_size: 11
    stride: 4
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
 }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "conv1"
  top: "conv1"
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm1"
  type: "LRN"
  bottom: "pool1"
  top: "norm1"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    lr_mult: 1
decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu2"
  type: "ReLU"
  bottom: "conv2"
  top: "conv2"
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "norm2"
  type: "LRN"
  bottom: "pool2"
  top: "norm2"
  lrn_param {
    local_size: 5
    alpha: 0.0001
    beta: 0.75
  }
}
layer {
  name: "conv3"
  type: "Convolution"
  bottom: "norm2"
  top: "conv3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
  name: "relu3"
  type: "ReLU"
  bottom: "conv3"
  top: "conv3"
}
layer {
  name: "conv4"
  type: "Convolution"
  bottom: "conv3"
  top: "conv4"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 384
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu4"
  type: "ReLU"
  bottom: "conv4"
}
layer {
  name: "conv5"
  type: "Convolution"
  bottom: "conv4"
  top: "conv5"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 1
    kernel_size: 3
    group: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu5"
  type: "ReLU"
  bottom: "conv5"
  top: "conv5"
}
layer {
  name: "pool5"
  type: "Pooling"
bottom: "conv5"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "fc6"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc6"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu6"
  type: "ReLU"
  bottom: "fc6"
  top: "fc6"
}
layer {
  name: "drop6"
  type: "Dropout"
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc7"
  type: "InnerProduct"
  bottom: "fc6"
  top: "fc7"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 4096
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layer {
  name: "relu7"
  type: "ReLU"
 bottom: "fc7"
  top: "fc7"
}
layer {
  name: "drop7"
  type: "Dropout"
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layer {
  name: "fc8"
  type: "InnerProduct"
  bottom: "fc7"
  top: "fc8"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 1000
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layer {
 name: "accuracy"
  type: "Accuracy"
  bottom: "fc8"
  bottom: "label"
  top: "accuracy"
  include {
    phase: TEST
  }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "fc8"
  bottom: "label"
  top: "loss"
}
I0706 11:00:31.481516  8842 layer_factory.hpp:74] Creating layer data
I0706 11:00:31.481535  8842 net.cpp:90] Creating Layer data
I0706 11:00:31.481547  8842 net.cpp:368] data -> data
I0706 11:00:31.481564  8842 net.cpp:368] data -> label
I0706 11:00:31.481578  8842 net.cpp:120] Setting up data
I0706 11:00:31.481590  8842 data_transformer.cpp:22] Loading mean file from: data/ilsvrc12/imagenet_mean.binaryproto
F0706 11:00:31.485407  8842 db_lmdb.hpp:13] Check failed: mdb_status == 0 (2 vs. 0) No such file or directory
*** Check failure stack trace: ***
    @     0x7f906fb6bdaa  (unknown)
    @     0x7f906fb6bce4  (unknown)
    @     0x7f906fb6b6e6  (unknown)
    @     0x7f906fb6e687  (unknown)
    @     0x7f906ff9453e  caffe::db::LMDB::Open()
    @     0x7f906feff412  caffe::DataLayer<>::DataLayerSetUp()
    @     0x7f906ff33d19  caffe::BasePrefetchingDataLayer<>::LayerSetUp()
    @     0x7f906ffafe33  caffe::Net<>::Init()
    @     0x7f906ffb1ba2  caffe::Net<>::Net()
    @     0x7f906ff6e034  caffe::Solver<>::InitTestNets()
    @     0x7f906ff6e72b  caffe::Solver<>::Init()
    @     0x7f906ff6e8f6  caffe::Solver<>::Solver()
    @           0x40c970  caffe::GetSolver<>()
    @           0x406621  train()
   @           0x404ab1  main
    @     0x7f906f07dec5  (unknown)
    @           0x40505d  (unknown)
    @              (nil)  (unknown)

**Comment by cuihenggang** [comment]
It seems like your LMDB data file directory is incorrect.

Cui

**Comment by cuihenggang** [comment]
db_lmdb.hpp is in the include/ directory (because it's a header file)

**Comment by andersonRocha091** [comment]
@cuihenggang Thank you for the heads up. I already found the problem, the lmdb files was generated in a totally different folder because I ran the createimagenet.sh as root of the system, but yesterday I was trying to run like a ordinary user, so the files was not availabe where it supposed to be. After copying the lmdb files back to the right place everything worked just fine. Thank you very much for the help and willingness
